### PR TITLE
React meow migration

### PR DESCRIPTION
**Pull request by howardt12345**
- Migrates meow and adds functionality that supports the existing meow capabilities
- Adds various new things
  - Adds channel name containing word matching
  - Adds random multiplier
  - Makes Reactions/Replies an Event type
    - Event has generic 
  - Changes the `match_link_id` system to a `condition` that can be specified
    - `condition` is a boolean condition that can use `criteria_link` values for a logical statement on whether an event should happen
- Various bugfixes
  - Fix resources not loading if an invalid JSON was present, skips the bad file
  - Enhances replying to consider criteria links
  - Fix logging

### CODE REVIEW

**Review by chanomkaimuk — CHANGES_REQUESTED** [review]

File: functions/reacts.py [hunk 1]
```diff
@@ -81,78 +90,183 @@ def check_resource_match(message_content: str, resource_name: str) -> bool:
...
+        return True
+
+    def link_id_exists(link_id):
+        return link_id in criteria_links
+    
+    # Parse the condition string into an AST
+    parsed_expr = ast.parse(condition, mode='eval')
+    
+    allowed_names = {
+        'or': or_,
```
[the remark this answers is not in the document]
Comment (howardt12345) [reply functions/reacts.py:247]: Why

[the remark this answers is not in the document]
Comment (chanomkaimuk) [reply functions/reacts.py:247]: `not` is pretty important I would say, like for "react to anything"

[the remark this answers is not in the document]
Comment (chanomkaimuk) [reply functions/reacts.py:247]: oh derp

**Review by chanomkaimuk — APPROVED** [review]
NOT (Approve NOR (NOT Approve))

**Pull request merged**